### DISCUSSION TITLE
How to implement the Charset Auto Selector

### DISCUSSION THREAD

**Discussion by DahyeonWoo** [euphony-io/euphony#239] [Q&A] [answered]
I want to implement the Charset Auto Selector, but I have a question about how to implement it.

1. Try encoding all ASCII charset, UTF16, UTF32, and UTF8 and find the shortest result.
2. Find the shortest string with Huffman encoding.

Is there any other way to implement it?

**Comment by designe** [comment, marked as answer]
Hi! @DahyeonWoo.
Thanks for leaving quesiton!

For Charset Auto Selector, 1st one is enough :)
`CharsetAutoSelector` will compare all charset's length. then, shortest charset will be return.

**Comment by DahyeonWoo** [comment]
Thank you for your advice on how to implement it.
I want to implement Huffman encoding in the future.
